### PR TITLE
Refactoring enabling async mail sending

### PR DESCRIPTION
**Pull request by wThomas84**

### CODE REVIEW

File: crowdsource-core/src/main/java/de/asideas/crowdsource/config/mail/MailSenderConfig.java [hunk 1]
```diff
@@ -36,6 +56,8 @@ public JavaMailSender javaMailSender() {
 
         Properties properties = new Properties();
         properties.setProperty("mail.smtp.starttls.enable", Boolean.toString(useStartTls));
```
Comment (CanardSauvage) [inline crowdsource-core/src/main/java/de/asideas/crowdsource/config/mail/MailSenderConfig.java]: Why do you do this? You already have the values via Injection and now you create a new Properties-Object as untyped value holder? You can always create a class "MailConfiguration" which would be way better.

**Comment by CanardSauvage** [comment]
What happens if the mail is not send because of an error? Then there will never be any mail, so you just send them async, but you can't guarantee, that they are send.

The overall quality is again good, so this can be merged.

**Comment by wThomas84** [comment]
It’s the same like before. Nothing is sent, operations/modifications to persistent objects aren’t rolled back.
Difference: The user doesn’t see an (misleading) error and we log the incident.

Von: Sebastian Waschnick <notifications@github.com<mailto:notifications@github.com>>
Antworten an: as-ideas/crowdsource <reply@reply.github.com<mailto:reply@reply.github.com>>
Datum: Donnerstag, 26. November 2015 um 10:23
An: as-ideas/crowdsource <crowdsource@noreply.github.com<mailto:crowdsource@noreply.github.com>>
Cc: Thomas Wendzinski <thomas.wendzinski@asideas.de<mailto:thomas.wendzinski@asideas.de>>
Betreff: Re: [crowdsource] Refactoring enabling async mail sending (#32)


What happens if the mail is not send because of an error? Then there will never be any mail, so you just send them async, but you can't guarantee, that they are send.

The overall quality is again good, so this can be merged.

**Comment by CanardSauvage** [comment]
Yes, but you could 
* add quartz scheduler 
* any job-framework
* or just save mails to send into the database and have a scheduler load mails to send from the database, send them and then delete them or try again.

**Pull request merged**